retro_pytorch/retrieval.py

```diff
@@ -292,7 +292,8 @@ def index_embeddings(
         metric_type = "l2",
         max_index_memory_usage = max_index_memory_usage,
         current_memory_available = current_memory_available,
-        should_be_memory_mappable = True,
+        make_direct_map = True
+        should_be_memory_mappable = False,
```

Comment:
oh you can actually keep the memory mapping, it will apply to everything except for the direct map

Reply:
I was getting the same error when I did this ☹️

Reply:
Can you send the full stack trace ? I don't understand

Reply:
Hey @rom1504 here is the full stack trace:

```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
[<ipython-input-3-7b2653026171>](https://localhost:8080/#) in <module>()
     32     knn_extra_neighbors = 100,                     # num extra neighbors to fetch
     33     max_index_memory_usage = '100m',
---> 34     current_memory_available = '1G'
     35 )
     36 

11 frames
[/usr/local/lib/python3.7/dist-packages/faiss/swigfaiss.py](https://localhost:8080/#) in reconstruct_n(self, i0, ni, recons)
   1997         :param recons:      reconstucted vector (size ni * d)
   1998         """
-> 1999         return _swigfaiss.Index_reconstruct_n(self, i0, ni, recons)
   2000 
   2001     def search_and_reconstruct(self, n, x, k, distances, labels, recons):

RuntimeError: Error in virtual void faiss::Index::reconstruct(faiss::Index::idx_t, float*) const at /project/faiss/faiss/Index.cpp:48: reconstruct not implemented for this type of index
```
You can use this colab to reproduce: https://colab.research.google.com/drive/1nvAxBGHnjrNFmgBm8Qu9ZRnmvwQyCFLI?usp=sharing

Reply:
Ok i found the problem. You installed faiss-gpu (first line of the colab). Don't do that, use faiss-cpu instead.
Then you can revert this PR